relentless/simulate/lammps.py

```diff
@@ -0,0 +1,120 @@
+        sim.type_map = {}
+        sim.typeid_map = {}
+        for i,t in enumerate(sim.ensemble.types):
+            sim.type_map[t] = i+1
+            sim.typeid_map[sim.type_map[t]] = t
+
+        return sim
+
+class LAMMPSOperation(simulate.SimulationOperation):
+    def __call__(self, sim):
+        cmds = self.to_commands(sim)
+        sim.lammps.commands_list(cmds)
+
+    @abc.abstractmethod
+    def to_commands(self, sim):
+        pass
+
+class Initialize(LAMMPSOperation):
+    def __init__(self, neighbor_buffer):
+        self.neighbor_buffer = neighbor_buffer
+
+    def extract_box_params(self, sim):
+        # cast simulation box in LAMMPS parameters
+        V = sim.ensemble.V
+        if V is None:
+            raise ValueError('Box volume must be set.')
+        elif not isinstance(V, TriclinicBox):
+            raise TypeError('LAMMPS boxes must be derived from TriclinicBox')
+
+        Lx = V.a[0]
+        Ly = V.b[1]
+        Lz = V.c[2]
+        xy = V.b[0]
+        xz = V.c[0]
+        yz = V.c[1]
+
+        lo = -0.5*np.array([Lx,Ly,Lz])
+        hi = lo + V.a + V.b + V.c
+        xlo = lo[0]
+        xhi = hi[0]
+        ylo = lo[1]
+        yhi = hi[1]
+        zlo = lo[2]
+        zhi = hi[2]
+
+        return np.array([xlo,xhi,ylo,yhi,zlo,zhi,xy,xz,yz])
+
+    def attach_potentials(self, sim):
+        cmds = ["neighbor {} multi".format(self.neighbor_buffer)]
+        cmds += ['pair_style table linear {}'.format(len(sim.potentials.pair.r))]
+        for i,j in sim.ensemble.pairs:
+            # these will need to be written into (potentially temporary) files and read in
+            # make sure to drop any entries with r = 0, since these are not allowed in lammps
+            pass
+        return cmds
+
+class InitializeRandomly(Initialize):
+    def __init__(self, neighbor_buffer, seed, units="lj", atom_style="atomic"):
+        super().__init__(neighbor_buffer)
+        self.seed = seed
+        self.units = units
+        self.atom_style = atom_style
+
+    def to_commands(self, sim):
+        cmds = ["units {}".format(self.units),
+                "boundary p p p",
+                "atom_style {}".format(self.atom_style)]
+
+        # make box from ensemble
+        box = self.extract_box_params(sim)
+        if not np.all(np.isclose(box[-3:],0)):
+            cmds += ["region box prism {} {} {} {} {} {} {} {} {}".format(*box)]
+        else:
+            cmds += ["region box block {} {} {} {} {} {}".format(*box[:-3])]
+        cmds += ["create_box {} box".format(len(sim.ensemble.types))]
+
+        # use lammps random initialization routines
+        for i in sim.ensemble.types:
+            cmds += ["create_atoms {typeid} random {N} {seed} box".format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i])]
```

Comment:
This is a pedantic change, but it makes sure that `seed` is actually used for the first type (rather than `seed+1`).
```suggestion
                                                                          seed=self.seed+sim.type_map[i]-1)]
```